app/models/manageiq/providers/vmware/infra_manager/provision/customization.rb

```diff
@@ -205,7 +206,10 @@ def load_customization_spec(custom_spec_name)
       raise MiqException::MiqProvisionError, "Customization Specification [#{custom_spec_name}] does not exist." if cs.nil?
       raise MiqException::MiqProvisionError, "Customization Specification [#{custom_spec_name}] for OS type [#{cs[:typ]}] does not match the template VM OS" if cs[:typ].downcase != source.platform
       _log.info "Using customization spec [#{cs.name}]"
-      return cs
+
+      source.with_provider_connection do |vim|
+        vim.getVimCustomizationSpecManager.getCustomizationSpec(cs.name)&.spec
```

Comment:
The CustomizationSpec has a number of properties where the VIM class matters (e.g.: `CustomizationLinuxOptions` vs `CustomizationWinOptions`) so we can't just use the standard hash stored in the DB

Previously this was a serialized VimHash which included the class names (we'll need a migration to clean these up, didn't realize that `CustomizationSpec#spec` was the full vim object)

Fetching the spec from the vcenter seemed to be the best option, other option is to add the VIM class name as a hash key when we convert from the native object to a hash in the parser as something like a `:_xsiType` key but the user would be able to modify those in the provision dialog unless we strip them out.

I'm open to other ideas though